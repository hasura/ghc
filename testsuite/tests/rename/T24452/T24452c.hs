-- Having Map.empty present, does not change the fact that Alternative.empty is not visible
{-# LANGUAGE GeneralizedNewtypeDeriving #-}
module BugReproduce where

import Control.Applicative (Alternative)
import qualified Data.Map as Map

newtype Foo a = MkFoo [a] deriving (Functor, Applicative)

instance Alternative Foo where
  empty = undefined
