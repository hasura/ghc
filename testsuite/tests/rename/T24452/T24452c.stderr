T24452c.hs:11:3: error: [GHC-54721]
    ‘empty’ is not a (visible) method of class ‘Alternative’